### ISSUE TITLE
long bindings subcrate rebuild on each run of 'cargo build'

### ISSUE THREAD

**Issue by milyin** [microsoft/winrt-rs#381]
Build result not cached after switch to new build scheme.
On project https://github.com/kennykerr/win2d-rs.git
First run:
```
>cargo build
   Compiling syn v1.0.54
   Compiling libc v0.2.81
   Compiling serde_derive v1.0.118
   Compiling serde v1.0.118
   Compiling serde_json v1.0.60
   Compiling num_cpus v1.13.0
   Compiling rayon-core v1.9.0
   Compiling rayon v1.5.0
   Compiling winmd-macros v0.8.0 (https://github.com/microsoft/winrt-rs#bcd0f5f7)
   Compiling winrt-gen-macros v0.8.0 (https://github.com/microsoft/winrt-rs#bcd0f5f7)
   Compiling winmd v0.8.0 (https://github.com/microsoft/winrt-rs#bcd0f5f7)
   Compiling winrt-gen v0.8.0 (https://github.com/microsoft/winrt-rs#bcd0f5f7)
   Compiling winrt-macros v0.8.0 (https://github.com/microsoft/winrt-rs#bcd0f5f7)
   Compiling winrt v0.8.0 (https://github.com/microsoft/winrt-rs#bcd0f5f7)
   Compiling bindings v0.1.0 (C:\Projects\Rust\win2d-rs.original\bindings)
   Compiling win2d-rs v0.1.0 (C:\Projects\Rust\win2d-rs.original)
    Finished dev [unoptimized + debuginfo] target(s) in 27.24s
```
Second and all subsequent runs:
```
   Compiling bindings v0.1.0 (C:\Projects\Rust\win2d-rs.original\bindings)
   Compiling win2d-rs v0.1.0 (C:\Projects\Rust\win2d-rs.original)
    Finished dev [unoptimized + debuginfo] target(s) in 0.84s
```

For larger projects it takes about 15 seconds for each run:
```
   Compiling bindings v0.1.0 (C:\Projects\Rust\minesweeper-rs\bindings)
   Compiling minesweeper-rs v0.1.0 (C:\Projects\Rust\minesweeper-rs)
    Finished dev [unoptimized + debuginfo] target(s) in 13.70s
```

**Comment by kennykerr** [comment]
I also noticed this, but I'm not yet sure what is causing the slowdown.

**Comment by milyin** [comment]
> I also noticed this, but I'm not yet sure what is causing the slowdown. It might have something to do with this not working correctly:
> 
> https://github.com/microsoft/winrt-rs/blob/bcd0f5f718e69956781bf27455c6cadad9a12f1f/crates/macros/src/lib.rs#L55

Yes, I confirm this. Commenting out this line fixes issue. Probably it's something about relative paths, checking

**Comment by kennykerr** [comment]
We could just remove this - it doesn't detect changes within the folder anyway so changes would likely require the developer to run `cargo clean` anyway.

**Comment by milyin** [comment]
If I replace relative path to absolute, it works ok and detects changes inside .windows directory. Will investigate it a bit more
```
          println!("cargo:rerun-if-changed=C:\\Projects\\Rust\\win2d-rs\\.windows");
 ```

**Comment by milyin** [comment]
When I copied .windows directory to "bindings" rebuilds stopped. This means that path to Cargo.toml of current project is base path when checking.
But relative path to "../.windows" doesn't work anyway. It seeps that cargo have some problem with relative paths in this context.
On the other hand there are lots of projects on github which uses absolute path in "cargo:rerun-if-changed" setting, so probably it's normal.
So I made request with this fix - use absolute path to ".windows": https://github.com/microsoft/winrt-rs/pull/382
